module/x/peggy/types/genesis.go

```diff
@@ -328,3 +335,17 @@ func strToFixByteArray(s string) ([32]byte, error) {
 	copy(out[:], s)
 	return out, nil
 }
+
+func validateInjContractAddress(i interface{}) error {
+	v, ok := i.(string)
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", i)
+	}
+	if err := ValidateEthAddress(v); err != nil {
+		// TODO: ensure that empty addresses are valid in params
+		if !strings.Contains(err.Error(), "empty") {
```

Comment:
Instead of parsing strings, replace `ValidateEthAddress` with a proper checks that are applicable to this context, i.e. a check that allows empty addresses.